### DISCUSSION TITLE
How to create a package with a build system that depends on the package version

### DISCUSSION THREAD

**Discussion by trahay** [spack/spack#29437] [Q&A] [answered]
I'm updating the `eztrace` package for the new eztrace release (version 2.0).

The current spack package is declared as an `AutotoolsPackage`, but since version 2.0 eztrace uses cmake.
I can update eztrace package as a  `CMakePackage`, but that would prevent from building eztrace version 1.x.

Is there a way to create an eztrace package (maybe using the `Package` class), that builds using autoconf for some versions (eg `version < 2`), or cmake for other versions (eg `version >= 2`) ?

**Comment by trahay** [comment, marked as answer]
I found the answer to my question in the Glog package:

Instead of declaring an `AutotoolsPackage`, I have to declare a  `Package`. Then , declare 2 `install` procedures and specify the version with `@when`.
Overall, the results looks like this:
```
class Eztrace(Package):
[...]
    def install(self, spec, prefix):
        configure('--prefix=%s' % prefix)
        make()
        make('install')

    @when('@2.0:')
    def install(self, spec, prefix):
        cmake_args = ['-DBUILD_SHARED_LIBS=TRUE']
        cmake_args.extend(std_cmake_args)

        with working_dir('spack-build', create=True):
            cmake('..', *cmake_args)
            make()
            make('install')
```

**Comment by adamjstewart** [comment]
See https://spack.readthedocs.io/en/latest/build_systems/multiplepackage.html for the current recommended approach. @alalazo is working on improving this, hopefully in the near future.
